### PR TITLE
AsciiDoc documentation of the @ReportEntry extension

### PR DESCRIPTION
**Pull request by Michael1993**
Depends on #173
---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/master/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

File: docs/report-entries.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+@Test
+@ReportEntry("foo")
+@ReportEntry("bar")
+void reportingTest() {
+    // YOUR TEST CODE HERE
+}
+----
+
+You can also declare your own keys in the annotation (if you don't declare it, JUnit
+will use the value of the entry as the key).
```
Comment (Bukama) [inline docs/report-entries.adoc]: Does ist really use the value of the entry as the key or does it use the string `value` as the key?

File: src/main/java/org/junitpioneer/jupiter/ReportEntry.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@ExtendWith(ReportEntryExtension.class)
+public @interface ReportEntry {
+
+	/**
+	 * Specifies the key of the pair that's to be published as a report entry.
+	 * Defaults to {@code "value"} and can't be blank.
+	 *
+	 * @see org.junit.jupiter.api.extension.ExtensionContext#publishReportEntry(String, String) ExtensionContext::publishReportEntry
+	 */
+	String key() default "value";
```
Comment (Bukama) [inline src/main/java/org/junitpioneer/jupiter/ReportEntry.java]: I mean this: The String "value" is used as default.

File: docs/report-entries.adoc (hunk 1 above)
Comment (Michael1993) [reply docs/report-entries.adoc]: You are right, the new solution does use the String "value" as a key if there is no key specified. My original solution used `publishReportEntry(String value)` <- that does behave this way.

I'll rewrite it.

File: src/main/java/org/junitpioneer/jupiter/ReportEntry.java (hunk 2 above)
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/ReportEntry.java]: Yes, even the unit tests show that you are correct.

File: docs/report-entries.adoc [hunk 3]
```diff
@@ -65,8 +65,8 @@ void reportingTest() {
 }
 ----
 
-You can also declare your own keys in the annotation (if you don't declare it, JUnit
-will use the value of the entry as the key).
+You can also declare your own keys in the annotation (if you don't declare it, the
+extension will use 'value' as the key).
```
Comment (Bukama) [inline docs/report-entries.adoc]: This should be correct. Thanks for editing it.

**Pull request merged**